routes/url-api-routes.js

```diff
@@ -32,4 +32,29 @@ module.exports = function(app) {
   //   });
   // });
 
+  // GET route for filtering by length
+    app.get("/api/urls/:pw/length", function(req, res) {
+      db.url.findAll({
+        include: [db.url],
+          where: {
+            db.user.sessionId: req.params.pw
+            db.url.length: {
+              lt: {
+              req.params.length
+            }
+          }
+      }).then(function(dbUser) {
+        res.json(dbUser);
+      });
+    });
+
+  // POST route to get all the URLs for the session ID and length
+      app.post("/api/urls/:pw/:length", function(req, res) {
+        db.User.create(req.body).then(function(dbUser) {
```

Comment:
db.user is lowercase above but uppercase here. only the lowercase will work with the way the models are written, but again, we can change that